### DISCUSSION TITLE
Fatture acquisti: API e webhook

### DISCUSSION THREAD

**Discussion by tommasinigiovanni** [fattureincloud/api#333] [Q&A] [answered]
ciao a tutti,
vi condivido una mia necessità per la quale sto trovando difficoltà:

Quando qualcuno ci manda una fattura, me la ritrovo in `Acquisti - Da registrare`
<img width="164" alt="image" src="https://github.com/fattureincloud/api/assets/1384416/8e435c53-e17a-46a6-8135-40c73899beba">

La mia necessità è che nel momento in cui l'amministrazione le registra, e le ritrovo in `Acquisti - Spese`, io riceva un alert in modo da poter prendere tutti i dati di questa nuova fattura e spostarli su monday

Ho pensato di utilizzare make.com 
Il problema è come configurare il trigger che scatti quando le fatture passano da `Acquisti - Da registrare` a  `Acquisti - Spese`.
Ho visto che il webhook `it.fattureincloud.webhooks.received_documents.update` non è ancora disponibile.
Qualcuno a qualche idea?

grazie mille

**Comment by emajo** [comment]
Ciao @tommasinigiovanni, i webhooks sul tipo` it.fattureincloud.webhooks.received_documents.update` non sono ancora attivi ma lo saranno nelle prossime settimane, intanto non so dirti se l'integrazione con make offra trigger diversi (come polling) in quanto è un integrazione di terze parti.

Grazie,
Emanuele.

**Reply by tommasinigiovanni** [reply]
@emajo grazie mille della risposta.
Mentre attendo con fermento il rilascio del webhook `it.fattureincloud.webhooks.received_documents.update` ho pensato di utilizzare l'endpoint `GET /c/{company_id}/received_documents` con `type: expense` ordinandolo per data e controllando ogni giorno se ci sono item con `updated_at` = `data odierna`
Può essere un buon approccio?

Un'altra domanda a corredo: con l'endpoint `GET /c/{company_id}/received_documents` se voglio filtrare gli item `Acquisti/spese`  e `Acquisti/Da registrare` devo guardare la proprietà `has_ts_pay_pending_payment` corretto?

grazie mille
Giovanni

**Reply by emajo** [reply]
Si, la tua soluzione può funzionare, per quanto riguarda invece le spese da registrare non dovrebbero proprio essere ritornate nella chiamata list received documents, ti risulta?

**Reply by tommasinigiovanni** [reply]
@emajo effettivamente mi sembra ci siano solo le `Acquisti/spese`
Invece per avere l'elenco delle `Da registrare` che endpoint devo utilizzare?
grazie mille

**Reply by emajo** [reply, marked as answer]
Ciao @tommasinigiovanni, attualmente non disponiamo di api per le spese da registrare.
